### ISSUE TITLE
Using a .gitignore file

### ISSUE THREAD

**Issue by thomaspeterflood** [fuchsiana/ThreeBlindMice#2]
A push from within a local repository, by default, includes everything in the repository. Some of these files are only for use on our local machines, such as IDE settings that should only apply to you and your IDE environment. 

This means we need to simultaneously have them in our local repository while excluding them from pushes to the shared repository. This is the job of a [.gitignore file](https://git-scm.com/docs/gitignore). 

It seems that the best approach is to [create a local .gitignore](https://help.github.com/articles/ignoring-files/#create-a-local-gitignore). It will be included in the ThreeBlindMice Repository and will therefore ignore the files that we choose to ignore. the file itself is just a plain text file with something like `*.DS_Store*`. which says ignore any file that contains the file extension .DS_Store. 

There is a lot of documentation on what to exclude depending on the type of project etc. for our purposes now I will add a local .gitignore to the ThreeBlindMice repo that ignores .DS_Store files. these are Mac settings to do with folders appearance etc. i.e not relevant to our repository; only relevant to the local machine on which they are created.

**Comment by neilerer** [comment]
Hi Thomas,
Just wondering how this file works, and have a few questions about it.
Do we have to move these files into the .gitignore file, or do we simply add '.DS_Store' to the end of already existing files we want ignored, and if so, is this added to the files in their current locations? 
How we know what should, or should not go into this file?

**Comment by thomaspeterflood** [comment]
the .gitignore file itself is a txt file and nothing else. the git configuration in the local repository (in this case ThreeBlindMice) routinely checks for a .gitignore file. If one is found it effectively scrapes the .gitignore asking is there anything in this repository I should ignore? by including `*.DS_Store*` in the .gitignore file we are effectively saying: ignore any files that contain `*.DS_Store*`. The reason for excluding these files in particular is that they are MAC OSX settings files that are specific to the machine on which they are generated and are therefore not relevant to the shared git repo. There are other types of files that we will need to filter in this way and we can simply add the appropriate text to the .gitignore file if and when we need. In the mean time the .gitignore file just remains where it is.

**Comment by fuchsiana** [comment]
Used https://www.gitignore.io/ to generate lists of all possible irrelevant files from Windows, MacOS, Python, Eclipse, and Jetbrains (Pycharm) that should be ignored and added them to the .gitignore file.  I don't think any of us are using any other applications yet that could generate irrelevant files (I've searched for Brackets and there was no list).  If any of us do start using another application in conjunction with the project, the site above can simply be searched and any additional list copied and pasted into the .gitignore file.

Note that initially amending gitignore didn't seem to do anything; I had to follow the steps described [here](http://blog.jonathanchannon.com/2012/11/18/gitignore-not-working-fixed/) in order for the changes to take effect.  This may need to be done again if the file is amended, but I'm not certain of this.

Closing issue as resolved.
